samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -94,19 +90,21 @@ public SamzaContainerController(
   public void startContainer(ContainerModel containerModel, Config config, int maxChangelogStreamPartitions) {
```

Comment:
Future clean-up: (pun intended ;) )

>> This method does not actually wait for the container to fully start-up
		
Would be nice to return a `Future` here. That way, the API looks more reasonable. Not sure what the scope of this will be. So, can defer it later.